Automattic/jetpack
# Add 'earn-newsletter' as a valid param to membership status endpint

Relates to https://github.com/Automattic/wp-calypso/issues/72282
Relates to https://github.com/Automattic/wp-calypso/pull/74146
Relates to D103932-code

## Proposed changes:

In order to display the correct notice in Calyso for Paid Newsletters, we need to pass the 'earn-newsletter' param to the memberships status endpoint. Currently the endpoint, doesn't support the 'earn-newsletter' string  


<img width="795" alt="Screenshot 2023-03-07 at 8 55 08 PM" src="https://user-images.githubusercontent.com/7076981/223599162-31b9f8aa-a703-48f9-aa20-e121307597ea.png">


### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






1. Add code to your sandbox by running `bin/jetpack-downloader test jetpack add/add-additional-string-param-for-newsletters`
2. Sandbox `pubic-api.wordpress.com` and a test site
3. Go to `https://wordpress.com/earn/payments/${site_slug}` and verify the `https://public-api.wordpress.com/wpcom/v2/sites/${site_id}/memberships/status?source=earn-newsletter` endpoint in the network tab returns a 200 response and valid data.
